infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/LogReplicationEvent.java

```diff
@@ -51,13 +52,16 @@ public enum LogReplicationEventType {
      */
     private LogReplicationEventMetadata metadata;
 
+    @Getter
+    private LogReplicationFSM replicationFsm;
+
```

Comment:
nit: would it be possible to utilize `LogReplicationContext` or the direct reference to `FsmTaskManager` to pass reference of FSMs? adding fsm reference to events seems to add some extra complexity and dependency

Reply:
we'll then need to maintain another data structure in the context then: session -> fsm.
And also have session in the event.

I think having a reference to fsm in events is fine..I see it as lesser complexity and maintenance. I read it as "this event is for this fsm".

Open for discussion though..

Reply:
I kind of agree with Chris's point.  My suggestion is: as these events get passed to the respective fsm itself using input(), no need to create them them with the fsm info.  In the fsm's input() method, we invoke taskManager.addTask(..., Fsm fsm).

Reply:
The challenge comes while processing the event...We need a way to map the event to the FSM they belong to now that all the sessions are sharing the thread pool.
The input path, I agree..we don't use the fsm in the event even now.

Reply:
discussed this offline. Going with the approach of passing the fsm to the process methods instead of having them in the events.